etcdctl/ctlv3/command/get_command.go

```diff
@@ -161,3 +170,55 @@ func getGetOp(cmd *cobra.Command, args []string) (string, []clientv3.OpOption) {
 
 	return key, opts
 }
+
+func getAllReadableWithRole(client *clientv3.Client, role string, result map[string]*pb.KeyValue) {
+	resp, err := client.Auth.RoleGet(context.TODO(), role)
+	if err != nil {
+		ExitWithError(ExitError, err)
+	}
+
+	for _, perm := range resp.Perm {
+		if perm.PermType == clientv3.PermWrite {
+			// skip write only permission
+			continue
+		}
+
+		opts := []clientv3.OpOption{}
+		opts = append(opts, clientv3.WithRange(string(perm.RangeEnd)))
+		getResp, err := client.KV.Get(context.TODO(), string(perm.Key))
```

Comment:
possibly inconsistent since there's no WithRev to fix the revision across multiple gets